TestClusterOf1UsingDiscovery doesn't happy if there is last release of etcd

### Which github workflows are flaking?

no

### Which tests are flaking?

TestClusterOf1UsingDiscovery

### Github Action link

no

### Reason for failure (if possible)

https://github.com/etcd-io/etcd/blob/8de14bd36e846eef7bc0d66924a28e1352bb5a72/tests/e2e/discovery_test.go#L56

When we enable `cluster_proxy` as tags, the [ProxyV2](https://github.com/etcd-io/etcd/blob/8de14bd36e846eef7bc0d66924a28e1352bb5a72/tests/framework/e2e/cluster_proxy.go#L209) doesn't have a chance to start because we just define it. And in our testing workflow `test-grpcproxy-e2e`, we don't download last release binary so that the case never run. It only happens in local test.


### Anything else we need to know?

If the ProxyV2 is not used any more, I think we should delete it and add the `!cluster_proxy` tag for the test.

Sorry for late reply. This issue was found when I was trying to upgrade grpc-gateway deps.
Sorry for the unclear comment.

The reproduce step is here

```bash
$ git checkout main
$ git clean -dxf
$ make
$ PASSES=release scripts/test.sh # download 3.5.x
$ cd tests
$ go test -v ./e2e -tags cluster_proxy -run TestClusterOf1UsingDiscovery
=== RUN   TestClusterOf1UsingDiscovery
    before.go:36: Changing working directory to: /tmp/TestClusterOf1UsingDiscovery908449421/001
    logger.go:146: 2024-03-30T16:39:35.713+0800 INFO    starting server...      {"name": "TestClusterOf1UsingDiscovery-test-0"}
    logger.go:146: 2024-03-30T16:39:35.713+0800 INFO    spawning process        {"args": ["/home/fuweid/workspace/etcd/bin/etcd-last-release", "--name=TestClusterOf1UsingDiscovery-test-0", "--listen-client-urls=http://localhost:2000", "--advertise-client-urls=http://localhost:2000", "--listen-peer-urls=http://localhost:2001", "--initial-advertise-peer-urls=http://localhost:2001", "--initial-cluster-token=new", "--data-dir", "/tmp/TestClusterOf1UsingDiscovery908449421/002", "--snapshot-count=10000", "--enable-v2", "--initial-cluster-token=new", "--initial-cluster=TestClusterOf1UsingDiscovery-test-0=http://localhost:2001", "--initial-cluster-state=new"], "working-dir": "/tmp/TestClusterOf1UsingDiscovery908449421/001", "name": "TestClusterOf1UsingDiscovery-test-0", "environment-variables": ["PATH=/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/mnt/c/Program Files (x86)/Microsoft SDKs/Azure/CLI2/wbin:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/WindowsPowerShell/v1.0/:/mnt/c/WINDOWS/System32/OpenSSH/:/mnt/c/Program Files/dotnet/:/mnt/c/Program Files/Go/bin:/mnt/c/Program Files/Git/cmd:/mnt/c/Users/weifu/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/weifu/AppData/Local/Programs/Microsoft VS Code/bin:/mnt/c/Users/weifu/go/bin:/home/fuweid/.fzf/bin:/usr/local/go/bin:/home/fuweid/go/bin:/opt/bin:/opt/fuwei/bin:/usr/local/go/bin:/home/fuweid/go/bin:/opt/bin:/opt/fuwei/bin", "ETCD_UNSUPPORTED_ARCH=amd64", "ETCD_VERIFY=all"]}
    logger.go:146: 2024-03-30T16:39:36.458+0800 INFO    started server. {"name": "TestClusterOf1UsingDiscovery-test-0", "pid": 15195}
    logger.go:146: 2024-03-30T16:39:36.458+0800 INFO    spawning process        {"args": ["/home/fuweid/workspace/etcd/bin/etcd-last-release", "grpc-proxy", "start", "--listen-addr", "localhost:2003", "--endpoints", "http://localhost:2000", "--advertise-client-url", "", "--data-dir", "/tmp/TestClusterOf1UsingDiscovery908449421/002"], "working-dir": "/tmp/TestClusterOf1UsingDiscovery908449421/001", "name": "TestClusterOf1UsingDiscovery-test-0", "environment-variables": ["PATH=/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/mnt/c/Program Files (x86)/Microsoft SDKs/Azure/CLI2/wbin:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/WindowsPowerShell/v1.0/:/mnt/c/WINDOWS/System32/OpenSSH/:/mnt/c/Program Files/dotnet/:/mnt/c/Program Files/Go/bin:/mnt/c/Program Files/Git/cmd:/mnt/c/Users/weifu/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/weifu/AppData/Local/Programs/Microsoft VS Code/bin:/mnt/c/Users/weifu/go/bin:/home/fuweid/.fzf/bin:/usr/local/go/bin:/home/fuweid/go/bin:/opt/bin:/opt/fuwei/bin:/usr/local/go/bin:/home/fuweid/go/bin:/opt/bin:/opt/fuwei/bin", "ETCD_UNSUPPORTED_ARCH=amd64", "ETCD_VERIFY=all"]}
    discovery_test.go:61: dial tcp 127.0.0.1:2002: connect: connection refused
    logger.go:146: 2024-03-30T16:39:36.480+0800 INFO    closing test cluster...
    logger.go:146: 2024-03-30T16:39:36.481+0800 INFO    stopping server...      {"name": "TestClusterOf1UsingDiscovery-test-0"}
    logger.go:146: 2024-03-30T16:39:36.491+0800 INFO    stopped server. {"name": "TestClusterOf1UsingDiscovery-test-0"}
    logger.go:146: 2024-03-30T16:39:36.491+0800 INFO    closing server...       {"name": "TestClusterOf1UsingDiscovery-test-0"}
    logger.go:146: 2024-03-30T16:39:36.491+0800 INFO    removing directory      {"data-dir": "/tmp/TestClusterOf1UsingDiscovery908449421/002"}
    logger.go:146: 2024-03-30T16:39:36.492+0800 INFO    closed test cluster.
--- FAIL: TestClusterOf1UsingDiscovery (0.78s)
FAIL
FAIL    go.etcd.io/etcd/tests/v3/e2e    0.795s
FAIL
```

I was thinking that this case doesn't work if there is last release binary. We should skip it instead of running.

--- a/tests/e2e/discovery_test.go
+++ b/tests/e2e/discovery_test.go
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build !cluster_proxy
+
 package e2e
 
 import (

